File: java/leetcode/SubarraySumEqualsK.java
package leetcode;

/*
* https://leetcode.com/problems/subarray-sum-equals-k/
*  
* */
public class SubarraySumEqualsK {
	public static void main(String[] args) {
		int[] nums = new int[] {1, 1, 1};
		int k = 2;
		subarraySum(nums, k);
	}
	
	static int subarraySum(int[] nums, int k) {
		int count = 0;
		for (int i=0; i<nums.length; i++) {
			int j = i;
			int sum = 0;
			while (j < nums.length) {
				sum += nums[j];
				if (sum == k) {
					count++;
				}
				j++;
			}
		}
        return count;
    }
}

File: javascript/subarray-sum-equals-k.js
/**
 * https://leetcode.com/problems/subarray-sum-equals-k/
 * 
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraySum = function(nums, k) {
    let count = 0;
    let sum = 0;
    nums.forEach((_, index) => {
        for (let i=index; i<nums.length; i++) {
            sum += nums[i];
            if (sum == k) {
                count++;
            }
        }
        sum = 0;
    });
    return count;
};